torchgeo/datasets/enviroatlas.py

```diff
@@ -0,0 +1,532 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""EnviroAtlas High-Resolution Land Cover datasets."""
+
+import os
+import sys
+from typing import Any, Callable, Dict, List, Optional, Sequence
+
+import fiona
+import matplotlib.pyplot as plt
+import numpy as np
+import pyproj
+import rasterio
+import rasterio.mask
+import shapely.geometry
+import shapely.ops
+import torch
+from matplotlib.colors import ListedColormap
+from rasterio.crs import CRS
+from torch import Tensor
+
+from .geo import GeoDataset
+from .utils import BoundingBox, download_url, extract_archive
+
+
+class EnviroAtlas(GeoDataset):
+    """EnviroAtlas dataset covering four cities with prior and weak input data layers.
+
+    The `EnviroAtlas
+    <https://doi.org/10.5281/zenodo.5778192>`_ dataset contains NAIP aerial imagery,
+    NLCD land cover labels, OpenStreetMap roads, water, waterways, and waterbodies,
+    Microsoft building footprint labels, high-resolution land cover labels from the
+    EPA EnviroAtlas dataset, and high-resolution land cover prior layers.
+
+    This dataset was organized to accompany the 2022 paper, `"Resolving label
+    uncertainty with implicit generative models"
+    <https://openreview.net/forum?id=AEa_UepnMDX>`_. More details can be found at
+    https://github.com/estherrolf/qr_for_landcover.
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * TODO
```

Comment:
Waiting on arxiv version of the paper

Reply:
Should we use the openreview link in the meantime or is the arxiv version coming soon?